pkg/controller/template/render.go

```diff
@@ -523,3 +524,13 @@ func urlPort(u string) (interface{}, error) {
 		return "", fmt.Errorf("unknown scheme in %s", u)
 	}
 }
+
+// escapeSystemdPercent is a template function that replaces the % character in a string to %%
+func escapeSystemdPercent(u string) (interface{}, error) {
+	percentReplacer := strings.NewReplacer("%", "%%")
+	if strings.Contains(u, "%") {
+		u = percentReplacer.Replace(u)
+	}
+
+	return u, nil
```

Comment:
Curious, why not?
```suggestion
	return strings.Replace(u, "%", "%%")
```
It seems much faster than allocating a new `func`, checking, then running the new `func.`